app-sre/qontract-reconcile
# Add integration to support Cloudflare DNS zone creation.

### Why:
To migrate from Dyn
### What:
New integration that can manage zones and records in Cloudflare. Based on this [design doc](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/design-docs/terraform-cloudflare-dns-support.md).
### Validation:

- Locally run the integration successfully. Test cases:
- Create zone, see it shows up in my test Cloudflare account only with pending name server update.
- Delete created zones
- Create A, CNAME, NS, MX records
- Delete any created records
- Update attributes of those records
- Create NS and MX records with multiple values with same name
- Tested integration manager plan to create multiple pod with zone name passed as sharding key
- Tested Sharded dry run
- Performance test: zone with 1000 record creation took ~5min and used ~100MB RAM.
### Dependency:
[schema changes
](https://github.com/app-sre/qontract-schemas/pull/367)
